src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -0,0 +1,115 @@
+module Dscp.Snowdrop.Expanders
+    ( expandBlock
+    , expandGTxs
+    ) where
+
+import Control.Lens (contramap)
+import Data.Default (Default)
+import qualified Data.List as L
+import qualified Data.Map.Strict as M
+import qualified Data.Set as S
+
+import qualified Snowdrop.Model.Block as SD
+import Snowdrop.Model.Expander (Expander (..), SeqExpanders (..), expandUnionRawTxs, mkDiffCS)
+import Snowdrop.Model.State.Accounting.Account (Account (..))
+import Snowdrop.Model.State.Core (ChgAccum, ChgAccumCtx, ERoComp, StateTxType (..), queryOne)
+import Snowdrop.Model.State.Restrict (RestrictCtx)
+import Snowdrop.Util
+
+import Dscp.Core
+import Dscp.Snowdrop.Configuration
+
+----------------------------------------------------------------------------
+-- Top-level expanders
+----------------------------------------------------------------------------
+
+-- | Expand block.
+expandBlock ::
+       ( Default (ChgAccum ctx)
+       , HasLens ctx RestrictCtx
+       , HasLens ctx (ChgAccumCtx ctx)
+       )
+    => Block
+    -> ERoComp Exceptions Ids Values ctx SBlock
+expandBlock Block{..} = SD.Block rbHeader <$> expandGTxs (rbbTxs rbBody)
+
+-- | Expand list of global txs.
+expandGTxs ::
+       ( Default (ChgAccum ctx)
+       , HasLens ctx RestrictCtx
+       , HasLens ctx (ChgAccumCtx ctx)
+       )
+    => [GTxWitnessed]
+    -> ERoComp Exceptions Ids Values ctx SPayload
+expandGTxs txs = expandUnionRawTxs getByGTx txs
+
+getByGTx ::
+       GTxWitnessed
+    -> ( StateTxType
+       , Proofs
+       , SeqExpanders Exceptions Ids Proofs Values ctx GTxWitnessed)
+getByGTx t@(GMoneyTxWitnessed tx) =
+    let (a, b) = toProofBalanceTx tx in (a, b, seqExpandersGTx t)
+
+seqExpandersGTx ::
+       GTxWitnessed -> SeqExpanders Exceptions Ids Proofs Values ctx GTxWitnessed
+seqExpandersGTx (GMoneyTxWitnessed _) =
+    contramap (\(GMoneyTxWitnessed tx) -> tx) seqExpandersBalanceTx
```

Comment:
It looks like its declared somewhere as
```haskell
newtype GMoneyTxWitnessed = GMoneyTxWitnessed <something>
```
without a name for retriever of `something`. What about adding that name there and using it here?

Reply:
(Unless its in snowdrop, ofc)